### PR TITLE
fix: add syslinux to create ISO

### PR DESCRIPTION
**Pull request by andrewrynhard**
We need syslinux to create an ISO.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2475)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -384,8 +384,10 @@ RUN apk add --no-cache --update \
     cdrkit \
     efibootmgr \
     qemu-img \
+    syslinux \
     util-linux \
     xfsprogs
+COPY --from=docker.io/autonomy/syslinux:v0.2.0-34-g7e57ef7 / /
```
Comment (smira) [inline Dockerfile:390]: should this be `v0.2.0-35-g8be6358` for the sake of consistency?

**Review by smira — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (andrewrynhard) [reply Dockerfile:390]: whoops, yes

**Review by rsmitty — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**